### ISSUE TITLE
"Replica assignment is invalid." sometimes seen when starting the canary.

### ISSUE THREAD

**Issue by k-wall** [strimzi/strimzi-canary#16]
The canary is manually co-deployed with strimzi.  Sometimes, when the canary starts up, the following message repeats in the logs.  

Restarting the canary doesn't clear the condition.

Also I notice that there's no metric counting these failures, so when this condition occurs, a team's SREs would be blind to the condition.

```
2020/12/11 12:36:51 The canary topic strimzi-canary already exists
2020/12/11 12:36:51 assignments = map[0:[0 1 2] 1:[1 2 0] 2:[2 0 1]], minISR = 2
2020/12/11 12:36:51 [[0 1 2] [1 2 0] [2 0 1]]
2020/12/11 12:36:51 Error altering topic strimzi-canary: kafka server: Replica assignment is invalid. - Inconsistent replication factor between partitions, partition 0 has 2 while partitions [2] have replication factors [3], respectively.
2020/12/11 12:36:51 ... reconcile done
```

```
curl --silent http://10.129.2.47:8080/metrics | grep strimzi_
# HELP strimzi_canary_records_produced_latency Records produced latency in milliseconds
# TYPE strimzi_canary_records_produced_latency histogram
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="0",le="100"} 0
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="0",le="200"} 1
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="0",le="400"} 1
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="0",le="800"} 1
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="0",le="1600"} 1
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="0",le="+Inf"} 1
strimzi_canary_records_produced_latency_sum{clientid="strimzi-canary-client",partition="0"} 182
strimzi_canary_records_produced_latency_count{clientid="strimzi-canary-client",partition="0"} 1
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="1",le="100"} 1
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="1",le="200"} 1
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="1",le="400"} 1
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="1",le="800"} 1
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="1",le="1600"} 1
strimzi_canary_records_produced_latency_bucket{clientid="strimzi-canary-client",partition="1",le="+Inf"} 1
strimzi_canary_records_produced_latency_sum{clientid="strimzi-canary-client",partition="1"} 3
strimzi_canary_records_produced_latency_count{clientid="strimzi-canary-client",partition="1"} 1
# HELP strimzi_canary_records_produced_total The total number of records produced
# TYPE strimzi_canary_records_produced_total counter
strimzi_canary_records_produced_total{clientid="strimzi-canary-client",partition="0"} 1
strimzi_canary_records_produced_total{clientid="strimzi-canary-client",partition="1"} 1
```

**Comment by k-wall** [comment]
@ppatierno FYI

**Comment by k-wall** [comment]
> Also I notice that there's no metric counting these failures, so when this condition occurs, a team's SREs would be blind to the condition.

@ppatierno I don't think this aspect was addressed. I think if ever we get an unexpected error condition (like this one) we should be incrementing an error count.  Would you agree?

**Comment by ppatierno** [comment]
@k-wall that error can't happen anymore because of the fix. I would agree that more in general we need proper and better error handling and it should be addressed working on #5
